### ISSUE TITLE
Cabbage widget to edit/save Csound code

### ISSUE THREAD

**Issue by gogins** [rorywalsh/cabbage#148]
I am requesting a new Cabbage widget, a text edit box that will enable users to edit the Csound source code of a plugin, and save it back to the plugin's CSD file. Ideally, the editor would have a _Save_ button, but alternatively, it would suffice to save the edited code every time the plugin is stopped, and to re-load the edited code every time the plugin is started.

**Comment by rorywalsh** [comment]
Why not open the .csd file in a proper editor and edit it there while the plugin is running? Add [`autoUpdate()`](https://cabbageaudio.com/docs/form/#autoUpdate) to your form declaration and it will automatically recompile each time you save changes. Note that if you add or remove widgets you will need to re-instantiate the plugin as widgets are tied to plugin parameters, and as you will know from your own experience, these cannot by altered dynamically while plugin is loaded 👍

**Comment by gogins** [comment]
I did not know about auto update, I will try that.

**Comment by gogins** [comment]
I tried this. I use Visual Studio code as an editor. First I tried with auto-save enabled in the editor. The Cabbage synth played as it should, but editing the .csd file caused Reaper to either freeze or crash. Next I tried with auto-save disabled, using a manual save after my edits. This casued Reaper to freeze or crash, but sometimes Reaper continued to run but the Cabbage synth went dead and would not play again until Reaper was closed or restarted.

I wll try again with a simpler Cabbage synth that I will post here.
